test/e2e/tests/test_db_instance.py

```diff
@@ -0,0 +1,132 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the RDS API DBInstance resource
+"""
+
+import boto3
+import datetime
+import logging
+import time
+from typing import Dict
+
+import pytest
+
+from acktest.k8s import resource as k8s
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_rds_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+
+RESOURCE_PLURAL = 'dbinstances'
+
+DELETE_WAIT_INTERVAL_SLEEP_SECONDS = 15
+DELETE_WAIT_AFTER_SECONDS = 120
+# NOTE(jaypipes): I've seen this take upwards of 5 minutes or more to fully see
+# the DB instance not appear in the DescribeDBInstances call once
+# DeleteDBInstance has been called (even with SkipFinalSnapshot=true)
+DELETE_TIMEOUT_SECONDS = 600
+
+CREATE_INTERVAL_SLEEP_SECONDS = 15
+# Time to wait before we get to an expected `available` state.
+CREATE_TIMEOUT_SECONDS = 600
+
+
+@pytest.fixture(scope="module")
+def rds_client():
+    return boto3.client('rds')
+
+
+@pytest.fixture(scope="module")
+def master_user_pass_secret():
+    ns = "default"
+    name = "dbsecrets"
+    key = "master_user_password"
+    secret_val = "secretpass123456"
+    k8s.create_opaque_secret(ns, name, key, secret_val)
+    yield ns, name, key
+    k8s.delete_secret(ns, name)
```

Comment:
question: How/Will this secret be deleted before tests end?

Reply:
yep, since this is a pytest fixture, it will be deleted after the smoke test that uses it completes (either successfully or fails)